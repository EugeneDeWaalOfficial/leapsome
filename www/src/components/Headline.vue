<template>
  <div class="headline">
    <div class="title-wrapper">
      <h2 class="title">{{ title }}</h2>
      <b-button class="button" variant="info" @click="$emit('btn-click')">{{
        buttonText
      }}</b-button>
    </div>
    <hr />
  </div>
</template>

<script>
export default {
  name: "Headline",
  props: {
    title: {
      type: String,
      required: true
    },
    buttonText: {
      type: String,
      required: true
    }
  },
  data() {
    return {};
  }
};
</script>

<style scoped>
.headline {
  margin-bottom: 3rem;
}

.title-wrapper {
  display: flex;
  justify-content: space-between;
}
</style>
