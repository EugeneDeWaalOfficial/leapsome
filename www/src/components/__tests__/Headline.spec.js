import { shallowMount } from "@vue/test-utils";

import Headline from "@/components/Headline.vue";

describe("Headline.vue", () => {
  it("renders title and button text when passed", () => {
    const title = "Some Title";
    const buttonText = "Some Button";

    const wrapper = shallowMount(Headline, {
      propsData: { title, buttonText }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
