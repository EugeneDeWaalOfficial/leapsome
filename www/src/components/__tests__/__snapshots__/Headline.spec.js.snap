// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline.vue renders title and button text when passed 1`] = `
<div
  class="headline"
>
  <div
    class="title-wrapper"
  >
    <h2
      class="title"
    >
      Some Title
    </h2>
     
    <b-button
      class="button"
      variant="info"
    >
      Some Button
    </b-button>
  </div>
   
  <hr />
</div>
`;
